### PR TITLE
fix: handle search result overflow

### PR DESCRIPTION
**Pull request by GiyoMoon**
Properly overflow search results on smaller screens/if many results are show.

https://github.com/user-attachments/assets/4ec76ad9-0c9b-4e6f-ae4e-34013f363ff0

### CODE REVIEW

File: src/ui/search.tsx [hunk 1]
```diff
@@ -113,141 +117,143 @@ export function Search() {
...
+								</button>
+							</Show>
 						</div>
-						<div class="mt-1 grow space-y-2 overflow-y-auto px-4 pt-2 scrollbar-thin">
-							<Suspense>
-								<Show
-									when={
-										searchTerm() &&
+					</div>
+					<div class="mt-1 lg:grow space-y-2 overflow-y-auto px-4 pt-2 scrollbar-thin">
```
Comment (femincan) [inline src/ui/search.tsx]:
```suggestion
					<div class="mt-1 lg:grow space-y-2 overflow-y-auto px-4 py-2 scrollbar-thin">
```

I think a small padding between the last search result item and bottom bar can be good.

Before:
image

After:
image

**Review by femincan — APPROVED** [review]
Everything looks okay!

File: src/ui/search.tsx (hunk 1 above)
Comment (GiyoMoon) [reply src/ui/search.tsx]: makes sense, updated it!

**Review by femincan — APPROVED** [review]

File: src/ui/search.tsx [hunk 2]
```diff
@@ -113,141 +117,143 @@ export function Search() {
...
-										</A>
-									</div>
-								</div>
-							</Suspense>
+													return (
+														<li ref={ref => setResultRefs(resultRefs => {
+															const newRefs = [...resultRefs];
+															newRefs[itemIndex] = ref;
+															return newRefs;
+														})} role="option" aria-selected={itemIndex === active() ? "true" : "false"}>
```
Comment (femincan) [inline src/ui/search.tsx]: Giving a `scroll-margin` to search result items when auto-scroll can create some space like the paddings. Currently, they are adjacent to the top or bottom side of the search results section.

```suggestion
														})} class="scroll-my-2" role="option" aria-selected={itemIndex === active() ? "true" : "false"}>
```

Before:

https://github.com/user-attachments/assets/83b519ef-2977-4393-9cbc-bd6faa50e91e

After:

https://github.com/user-attachments/assets/70814acd-1fad-4576-bfc7-da9ad1cde6b3

**Review by LadyBluenotes — APPROVED** [review]

**Pull request merged**